protocol/src/protocol.resourceChanges.proposed.md

```diff
@@ -0,0 +1,82 @@
+# Resource Changes
+
+_Client Capabilities_:
+
+The client sets the following capability if it is supporting resource Changes.
+
+```ts
+/**
+ * The workspace client capabilities
+ */
+workspace: {
+  workspaceEdit?: {
+		  /**
+		  * The client supports resource changes in `WorkspaceEdit`s
+		  */
+		  resourceChanges?: boolean;
+	  };
+}
+```
+
+_WorkspaceEdit_
+
+The following additions are proposed for __WorkspaceEdit__.
+
+
+_resourceChanges_
+
+A workspace edit can optionally provide resource changes. If present resource changes are applied before the `changes/documentChanges` has been applied
+allowing the `change/documentChanges` to work on affected files. The changes are applied in the order that they are supplied.
+
+```ts
+interface  WorkspaceEdit{
+
+  resourceChanges?: ResourceChange[]
+
+}
+```
+
+## ResourceChange
+Changes to a folder or file on the workspace.
+
+```ts
+export interface ResourceChange{
+
+  /**
+  * The resource change operation type
+  */
+  changeType: ResourceChangeType
+
+  /**
+  * The resource.
+  *
+  * If change type is CREATE a new file is created on the uri.
+  * This operation will not overwrite any existing resources and will fail.
+  *
+  * If change type is DELETE the resource is deleted if it exists. If the
+  * uri is a folder this operation deletes all its children.
+  *
+  * If change type is MOVE this uri is considered the source resource for the
+  * move and must point to an existing resource
+  */
+  path: string
+
+  /**
+  * The new uri for the resource pointed by path.
+  * Required only if change type is MOVE. Must be compatible with the path
+  * uri ie. must be a file uri if path is a file uri.
+  *
+  */
+  newPath?: string
+
+}
+
+/**
+ * A resource change type
+ */
+export declare namespace ResourceChangeType{
```

Comment:
Should add a space after ResourceChangeType to be consistent with the other files.